packages/blocks/src/api/utils.js

```diff
@@ -44,11 +44,19 @@ export function isUnmodifiedBlock( block, role ) {
 
 	// Filter attributes by role if a role is provided.
 	const attributesToCheck = role
-		? Object.entries( blockAttributes ).filter(
-				( [ , definition ] ) =>
+		? Object.entries( blockAttributes ).filter( ( [ key, definition ] ) => {
+				// A special case for the metadata attribute.
+				// It can include block bindings that serve as a source of content,
+				// without directly modifying content attributes.
+				if ( role === 'content' && key === 'metadata' ) {
+					return true;
```

Comment:
We can be more granular here, but then it would mean we have to keep track of every new `metadata` sub-attribute.

```js
return (
  Object.keys( block.attributes[ key ].bindings ?? {} )
	  .length > 0
);
```